File: src/java/com/wasteinformationserver/website/HttpTools.java
package com.wasteinformationserver.website;

import java.math.BigInteger;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/**
 * basic http tools
 *
 * @author Lukas Heiligenbrunner
 */
public class HttpTools {
    /**
     * create md5 hash of string
     *
     * @param value input string
     * @return md5 hash
     */
    public static String StringToMD5(String value) {
        try {
            MessageDigest md = MessageDigest.getInstance("MD5");
            byte[] messageDigest = md.digest(value.getBytes());
            BigInteger no = new BigInteger(1, messageDigest);
            return no.toString(16);
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
            return "";
        }
    }
}

File: src/java/com/wasteinformationserver/website/HttpTools.kt
package com.wasteinformationserver.website

import java.math.BigInteger
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

/**
 * basic http tools
 *
 * @author Lukas Heiligenbrunner
 */
class HttpTools {
    companion object{
        /**
         * create md5 hash of string
         *
         * @param value input string
         * @return md5 hash
         */
        fun StringToMD5(value: String): String {
            return try {
                val md = MessageDigest.getInstance("MD5")
                val messageDigest = md.digest(value.toByteArray())
                val no = BigInteger(1, messageDigest)
                no.toString(16)
            } catch (e: NoSuchAlgorithmException) {
                e.printStackTrace()
                ""
            }
        }
    }
}